internal/pkg/sanitize/sanitize.go

```diff
@@ -36,33 +36,34 @@ func Sanitize(pkgPattern string, opts *Options) (*packages.OverlayJSON, error) {
 	}
 	log.Debugf("Working directory: %q\n", workDir)
 
-	for _, pkg := range pkgs {
+	packages.Visit(pkgs, nil, func(pkg *packages.Package) {
 		if !opts.shouldSanitize(pkg.PkgPath) {
-			continue
+			return
 		}
+		log.Debugf("Instrumenting package %q", pkg.PkgPath)
 		for i, sourceFile := range pkg.Syntax {
 			originalSourceFile := pkg.CompiledGoFiles[i]
 			transformer := hook.NewTransformer(sourceFile, pkg.Fset, originalSourceFile, pkg.TypesInfo, hook.NodeId)
 			if numHooks := transformer.TransformFile(); numHooks > 0 {
-				rel, err := fileutil.RelativePathInModule(pkg.Module.Dir, originalSourceFile)
+				baseFile := filepath.Base(originalSourceFile)
+				baseFileNoExtension := baseFile[:len(baseFile)-len(filepath.Ext(baseFile))]
+				instrumentedFile, err := os.CreateTemp(workDir, fmt.Sprintf("%s-*.go", baseFileNoExtension))
 				if err != nil {
-					log.Warnf("Skipped instrumenting %q: %s", originalSourceFile, err.Error())
+					log.Warnf("Failed to create a temporary file to store the instrumented code")
```

Comment:
Could it be helpful to know for which `sourceFile` we're doing an early loop continue? Additional question, if we fail here couldn't that also be a reason for a process abortion? Or do we want to accept partial instrumentations (due to a failure in tmp-file creation?)

Reply:
Sounds reasonable to me as long as the user is informed about potential issues with partial/missing instrumentation. Do you think it's reasonable to just use the warnings as is as an indicator or should we output a more concrete information at the very end of the instrumentation process in these cases. Something like: 

```
Note: Not all files could be instrumented as expected. Performance could be impacted. 
```

Reply:
Added warnings about the effect of skipping instrumentation for source files